mdn/yari
# sign in and sign up

Part of #2449

This depends on https://github.com/mdn/kuma/pull/7794
Eventually, we'll need to merge and upgrade Kuma first but the eager ones can use my branch there locally. 

Also, to be able to test this:

- Add `REACT_APP_USE_YARI_SIGNIN=true` in your `.env`
- Go to <http://localhost.org:3000> and click Sign-in

Stuff to look over before turning this into a real PR:

- [x] Document, in Yari, how this stuff works
- [x] Manually test signing up with a GitHub user that has never signed up before
- [x] That `next` URL that gets set should *always* be present but if it becomes something like `/en-US/signin` or `/en-US/signup` it could leave you in a weird loop. 
- [ ] ~The sign-up page still feels a bit fragile and weird since it depends on a complex redirect dance with Kuma. At the bottom of the sign-up page it should have a "support link" to file a new yari github issue. The old Kuma `signup.html` has a link to file a new bugzilla ticket.~ This can be discussed as the sign up page gets some improved styling
- [x] After you've signed up, the `whoami` isn't "re-fetched" so it still says "Sign in" when you're redirected after having created the account

Note that the sign-up form only asks 1 single question. Just `terms`. Nothing about newsletters. That's right, right?
